### PR TITLE
fix duplication spinner issue

### LINKED ISSUE #3230: Duplication spinner goes on forever
Investigate why the background process doesn't signal its completion.
Possible that the frontend loops crashes if one of the refreshes times out.

### PR DESCRIPTION
**Pull request by ekowidianto**
Fixes #3230 

- Instead of having rails to show spinner and refresh, react frontend now queries backend if a job is done or not
- Added a feature to send an email when a course duplication fails

### CODE REVIEW

File: client/app/bundles/course/duplication/reducers/duplication.js [hunk 1]
```diff
@@ -112,10 +113,10 @@ export default function (state = initialState, action) {
     }
 
     case actionTypes.DUPLICATE_COURSE_REQUEST:
-    case actionTypes.DUPLICATE_ITEMS_REQUEST: {
+    case actionTypes.DUPLICATE_ITEMS_REQUEST:
+    case actionTypes.DUPLICATE_COURSE_SUCCESS: {
```
Comment (zhuhanming) [inline client/app/bundles/course/duplication/reducers/duplication.js:117]: Hmm this change doesn't make sense to me. If it succeeded, the isDuplicating should be `false`?

Comment (ekowidianto) [reply client/app/bundles/course/duplication/reducers/duplication.js:117]: Actually there is a small time gap between duplication success and redirection to the new course. So to keep the circular progress going, I set the isDuplicating to true. Let me know what you think.

Comment (zhuhanming) [reply client/app/bundles/course/duplication/reducers/duplication.js:117]: I think if that's the case, there should be a third state, perhaps tracked by another `isSuccess` boolean.

Then, upon succeeding, the spinner changes to a message like "Successfully duplicated. Redirecting you to the new course now...", with the button still disabled.

Comment (ekowidianto) [reply client/app/bundles/course/duplication/reducers/duplication.js:117]: haha right. actually I have those - let me change it!

**Review by zhuhanming — APPROVED** [review]

**Pull request merged**